pkg/infra/iac2/templates/rds_instance/factory.ts

```diff
@@ -0,0 +1,38 @@
+import * as aws from '@pulumi/aws'
+import * as fs from 'fs'
+
+interface Args {
+    Name: string
+    SubnetGroup: aws.rds.SubnetGroup
+    SecurityGroups: aws.ec2.SecurityGroup[]
+    IamDatabaseAuthenticationEnabled: boolean
+    DatabaseName: string
+    Engine: string
+    EngineVersion: string
+    InstanceClass: string
+    SkipFinalSnapshot: boolean
+    AllocatedStorage: number
+    protect: boolean
+    CredentialsPath: string
+}
+
+// noinspection JSUnusedLocalSymbols
+function create(args: Args): aws.rds.Instance {
+    return new aws.rds.Instance(
+        args.Name,
+        {
+            instanceClass: args.InstanceClass,
+            engine: args.Engine,
+            engineVersion: args.EngineVersion,
+            dbName: args.DatabaseName,
+            username: fs.readFileSync(args.CredentialsPath, 'utf-8').split('\n')[1].split('"')[3],
+            password: fs.readFileSync(args.CredentialsPath, 'utf-8').split('\n')[2].split('"')[3],
```

Comment:
I wonder, will ts let us use `...` notation to turn this into an invocation of an anonymous function? something like:

```
/* ... */
    dbName: args.DatabaseName,
    ... () => { // literally the dots
      const fileContents = fs.readFileSync(args.CredentialsPath, 'utf-8');
      /* ... */
      return usernameContents, passwordContents;
    }(),
},
```

Reply:
so i had tried to do something like that, but couldnt get it to be happy in our formatting/unit tests. I can keep trying though